### PR TITLE
Add utilities to simplify IAST smoke testing

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Updates IAST smoke tests to reuse logic in common base class

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy [hunk 1]
```diff
@@ -130,8 +123,10 @@ class IastSpringBootSmokeTest extends AbstractSpringBootIastTest {
     then:
     response.isSuccessful()
     response.header("Set-Cookie").contains("user-id")
-    waitForSpan(new PollingConditions(timeout: 5),
-    hasVulnerability(type('INSECURE_COOKIE').and(evidence('user-id'))))
+    hasVulnerabilityInLogs { vul ->
```
Comment (jandro996) [inline dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy:126]: It's necessary to change hasVulnerability for hasVulnerabilityInLogs? #2 😄

File: dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy [hunk 2]
```diff
@@ -145,8 +140,10 @@ class IastSpringBootSmokeTest extends AbstractSpringBootIastTest {
...
```
Comment (jandro996) [inline dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy:143]: It's necessary to change hasVulnerability for hasVulnerabilityInLogs? #3 😃

File: dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy (hunk 1 above)
Comment (manuel-alvarez-alvarez) [reply dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy:126]: Same as before

File: dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy (hunk 2 above)
Comment (manuel-alvarez-alvarez) [reply dd-smoke-tests/springboot/src/test/groovy/datadog/smoketest/IastSpringBootSmokeTest.groovy:143]: Same :laughing:

**Review by jandro996 — APPROVED** [review]

**Review by smola — APPROVED** [review]

**Pull request merged**